app/src/hooks/useBalance.tsx

```diff
@@ -62,24 +61,18 @@ const useBalance = ({ env, api, chain, token, address }: UseBalanceParams) => {
         }
 
         case Network.Polkadot: {
-          if (getNativeToken(chain).id === token.id) {
-            const result = await getNativeBalance(api, address)
-
-            fetchedBalance = {
-              value: result?.data.free || 0n,
-              formatted: toHuman(result?.data.free || 0n, token).toString(),
-              decimals: token.decimals,
-              symbol: token.symbol,
-            }
-          } else {
-            const result = await getNonNativeBalance(api, chain, token, address)
-
-            fetchedBalance = {
-              value: result?.free || 0n,
-              formatted: toHuman(result?.free || 0n, token).toString(),
-              decimals: token.decimals,
-              symbol: token.symbol,
-            }
+          const relay = getRelayNode(env)
+          const node = assets.getTNode(chain.chainId, relay)
+          if (!node) throw new Error('Node not found')
+
+          const tokenSymbol = getTokenSymbol(node, token)
+          const balance = await getAssetBalance(address, node, { symbol: tokenSymbol })
```

Comment:
NIT but I'd add the fallback here so you don't have to repeat it down below
```suggestion
          const balance = (await getAssetBalance(address, node, { symbol: tokenSymbol })) ?? 0
```